railties/lib/rails/application.rb

```diff
@@ -386,7 +386,8 @@ def config=(configuration) #:nodoc:
     def secrets
       @secrets ||= begin
         secrets = ActiveSupport::OrderedOptions.new
-        secrets.merge! Rails::Secrets.parse(config.paths["config/secrets"].existent, env: Rails.env)
+        secrets.merge! Rails::Secrets.parse(config.paths["config/secrets"].existent, env: Rails.env,
```

Comment:
How about we just filter out the encrypted secrets file instead:

```ruby
files = config.paths["config/secrets"].existent
files = files.reject { |path| path.end_with?(".enc") } unless config.read_encrypted_secrets
secrets.merge! Rails::Secrets.parse(files, env: Rails.env)
```

Then `preprocess` becomes (which will barf if the key is missing):

```ruby
def preprocess(path)
  if path.end_with?(".enc")
    decrypt(IO.binread(path))
  else
    IO.read(path)
  end
end
```